spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubAcknowledgmentCallback.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.integration.inbound;
+
+import org.springframework.cloud.gcp.pubsub.integration.AckMode;
+import org.springframework.cloud.gcp.pubsub.support.AcknowledgeablePubsubMessage;
+import org.springframework.integration.acks.AcknowledgmentCallback;
+
+/**
+ * Pub/Sub specific implementation of {@link AcknowledgmentCallback}.
+ *
+ * <p>{@link AcknowledgmentCallback#isAcknowledged()} is implemented for semantic accuracy
+ * only, since in Pub/Sub acknowledging a message more than once is not an error.
+ * <p>{@link AcknowledgmentCallback#noAutoAck()} is not implemented; the correct way to
+ * enable manual acking is through configuring {@link PubSubMessageSource}.
+ *
+ * @author Elena Felder
+ *
+ * @since 1.2
+ */
+public class PubSubAcknowledgmentCallback implements AcknowledgmentCallback {
+
+	private AcknowledgeablePubsubMessage message;
+
+	private AckMode ackMode;
+
+	private boolean acknowledged;
+
+	public PubSubAcknowledgmentCallback(AcknowledgeablePubsubMessage message, AckMode ackMode) {
+		this.message = message;
+		this.ackMode = ackMode;
```

Comment:
Both these properties must be `final` and probably with `Assert.notNull()` here in the ctor...